client/App.js

```diff
@@ -106,16 +100,45 @@ export default class App extends React.Component {
       'SF Pro Text Bold': require('./assets/fonts/SFProText/SF-Pro-Text-Bold.ttf'),
     })
 
+    await registerForPushNotificationsAsync();
+
+    // Push notifications listener <- the observer!
+    this.listener = Notifications.addListener(this.handleNotification);
+
     // App is ready to be loaded.
     this.setState({ appHasLoaded: true });
 
   }
 
+  // Callback to run when we observe a new
+  // push notification
+  handleNotification = (notification) => {
+    this.setState({ notification });
+  };
+
+  // Renders the notification component
+  // with the proper notification data
+  // If no notification data is passed in, render nothing.
+  NotificationContainer = () => {
+    if (!this.state.notification.data) return null;
```

Comment:
Yeah, it's apparently required I return `null` because `NotificationContainer` is a component. (I originally had it as `undefined` and React gave me the red screen telling me to change it :joy:)